executor/builder.go

```diff
@@ -825,11 +825,43 @@ func (b *executorBuilder) buildHashJoin(v *plan.PhysicalHashJoin) Executor {
 	return e
 }
 
+// wrapCastForAggArgs wraps the args of an aggregate function with a cast function.
+func (b *executorBuilder) wrapCastForAggArgs(funcs []*aggregation.AggFuncDesc) {
```

Comment:
We'd better make implement a function like `InferArgumentType` for `AggFuncDesc`, and call that function inside this routine:
```go
for _, aggFuncDesc := range funcs {
    aggFuncDesc.InferArgumentType()
}
```

Reply:
I think current function name may be clearer,
since this is not a `real` **type infer**.
Type infer for agg function has been done in [AggFuncDesc.typeInfer](https://github.com/pingcap/tidb/blob/master/expression/aggregation/descriptor.go#L97-L114).
This function is more like a **supplementary behavior** for type infer.